_posts/admin/2018-09-25-custom-sizing-of-services.md

```diff
@@ -0,0 +1,48 @@
+---
+date: 2018-09-25
+title:  Applying Custom Sizes to Services
+categories:
+    - Admin
+description:  Applying Custom Sizes to Services
+type: Document
+---
+
+_Note: This guide assumes you're deploying Spinnaker on Kubernetes using the [Distributed](https://www.spinnaker.io/setup/install/environment/#distributed-installation) deployment type with Halyard._
+ ***
+ You may find the need to apply custom limits and requests to some of the Spinnaker services -- doing so can provide more stable behavior of the services while also preventing extreme bursts of CPU activity or the allocation of more memory than is available (and thus being OOM Killed).
+
+ ### Custom Sizing in Hal Config
+ Custom sizing can be applied to a service through the `deploymentEnvironment.customSizing` section of your hal config. The following example sets requests for `gate` of 250 millicpu and 512 MiB with limits of 500 millicpu and 1GiB of memory.
+ ```
+ deploymentEnvironment:
+  customSizing:
+    gate:
+      limits:
+        cpu: 500m
+        memory: 1Gi
+      requests:
+        cpu: 250m
+        memory: 512Mi
+```
+ (Note: for these settings to be applied to sidecar containers as well, prefix the service name with "spin-")
+
+ For a more complete understanding of these settings see the [Kubernetes docs](https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/)
+ 
+ *** 
+ ### JAVA_OPTS in Service Settings
```

Comment:
You've got leading spaces on a bunch of this stuff, it's preventing the formatting from taking effect.

Reply:
https://cl.ly/bf7b505e4bb0/Image%202018-09-26%20at%209.34.09%20AM.png